ui/common/src/storage.ts

```diff
@@ -67,20 +67,26 @@ export const storedJsonProp =
   };
 
 export interface StoredMap<V> {
-  (key: string): V;
-  (key: string, value: V): void;
+  getMap(): Map<string, V>;
+  getKey(key: string): V;
+  setKey(key: string, value: V): void;
 }
 
 export const storedMap = <V>(propKey: string, maxSize: number, defaultValue: () => V): StoredMap<V> => {
   const prop = storedJsonProp<[string, V][]>(propKey, () => []);
   const map = new Map<string, V>(prop());
-  return (key: string, v?: V) => {
-    if (defined(v)) {
+  return {
+    getMap: () => {
+      return map;
+    },
+    getKey: (key: string) => {
+      const ret = map.get(key);
+      return defined(ret) ? ret : defaultValue();
+    },
+    setKey: (key: string, v: V) => {
       map.delete(key); // update insertion order as old entries are culled
       map.set(key, v);
       prop(Array.from(map.entries()).slice(-maxSize));
-    }
-    const ret = map.get(key);
-    return defined(ret) ? ret : defaultValue();
+    },
```

Comment:
We can't use ES6 sets as a `StoredJsonProp` since they're not `JSON.stringify`-able and it seems wasteful to convert them back and forth into arrays for storage. 

Instead, I'm making some small refactors to @schlawg's `StoredMap` so I can use it to keep a set of users in local storage. This was added fairly recently iirc so seems it's safe to change. Again lmk if there's a cleaner approach.

Reply:
The maps can't be `JSON.stringify`'d all the same though and are converted to and from arrays, no? Using a set version definitely seems like a cleaner solution.

Reply:
The tiny bit of code in StoredMapProperty was originally purposed to rely on insertion order to store toggle button values for a window of recent studies.  It was moved into storage.ts in the merge, but it is not currently a general purpose map implementation at all.  It was designed for very narrow circumstances.

It might not be a good idea to expose the underlying map with getMap() as you risk client code ignoring setKey and just updating the map directly.  They could mess up the insertion order by updating an existing map entry, which is a problem if it ever grows beyond maxSize because the element ordering informs what to slice.

Benedikt suggests serializing sets to/from JSON.  I'd do that.  And hey don't sweat the array conversion overhead.

Reply:
Makes sense, will do. Appreciate the feedback